Add integration tests for Ambassador 0.86.1

concaf opened the pull request:
## Proposed Changes

* This commit adds `--ambassador-version` switch to e2e tests to run integration tests with Ambassador.

**Release Note**

NONE

mattmoor commented on third_party/ambassador-0.86.1/README.md:
```diff
@@ -0,0 +1,5 @@
+
+`ambassador-rbac.yaml` and `ambassador-service.yaml` are required to install Ambassador. These need to be updated every
+every time Ambassador's version is bumped. Detailed instructions on installing Ambassador are available
+[here](https://www.getambassador.io/user-guide/getting-started/).
```
<!--whitespace-bot-->
Add trailing newline:
```suggestion
[here](https://www.getambassador.io/user-guide/getting-started/).

```

concaf commented:
@googlebot I signed it!

concaf commented:
@googlebot I fixed it.

concaf commented:
@googlebot I fixed it.

concaf commented:
@mattmoor @bbrowning I'd love to get an `/ok-to-test` here so I can identify problems with the integration and fix accordingly. Thanks.

mattmoor commented:
/ok-to-test

We may also need a prow job before this does anything (cc @chaodaiG )

concaf commented:
@mattmoor @chaodaiG I've sent https://github.com/knative/test-infra/pull/1598 as a prow job required for this PR. I'm not familiar with Prow at all, so looking forward to feedback there :see_no_evil:

chizhg commented on test/e2e-common.sh:
```diff
@@ -191,6 +199,23 @@ function install_kourier() {
 }
 
+function install_ambassador() {
+  AMBASSADOR_MANIFESTS_PATH="./third_party/ambassador-0.86.1/"
```
Will it be better to follow the naming convention and change the folder name to `ambassador-latest`?

concaf replied:
Fixed in next commit. Thanks!

mattmoor commented:
This likely needs an `INGRESS_CLASS` in `test/e2e-common.sh` now.  We've also been adding a lot of Ingress conformance tests to make it easier to identify gaps and fix them.

concaf commented:
> This likely needs an `INGRESS_CLASS` in `test/e2e-common.sh` now. We've also been adding a lot of Ingress conformance tests to make it easier to identify gaps and fix them.

@mattmoor I've rebased to current master and added the `INGRESS_CLASS` as well. Thanks.

mattmoor commented:
@chaodaiG @adrcunha What needs to happen to validate this?

mattmoor commented on test/e2e-common.sh:
```diff
@@ -261,6 +290,22 @@ data:
+    serving.knative.dev/release: devel
+data:
+  ingress.class: "ambassador.ingress.networking.knative.dev"
+  clusteringress.class: "ambassador.ingress.networking.knative.dev"
```
Perhaps these should key off of `${INGRESS_CLASS}`?

I hadn't seen this previously or I'd have done this when I added it :)

concaf replied:
Fixed!

adrcunha commented on test/e2e-common.sh:
```diff
@@ -195,6 +204,23 @@ function install_kourier() {
+  kubectl apply -n ambassador -f ${AMBASSADOR_MANIFESTS_PATH} || return 1
+
+  echo ">> Fixing Ambassador's permissions"
+  kubectl patch clusterrolebinding ambassador -p '{"subjects":[{"kind": "ServiceAccount", "name": "ambassador", "namespace": "ambassador"}]}'
```
```suggestion
  kubectl patch clusterrolebinding ambassador -p '{"subjects":[{"kind": "ServiceAccount", "name": "ambassador", "namespace": "ambassador"}]}' || return 1
```

adrcunha commented on test/e2e-common.sh:
```diff
@@ -191,6 +199,23 @@ function install_kourier() {
 }
 
+function install_ambassador() {
+  AMBASSADOR_MANIFESTS_PATH="./third_party/ambassador-0.86.1/"
```
Outdaded, ignore.

concaf commented:
I've updated the PR with the requested changes.

adrcunha commented on test/e2e-common.sh:
```diff
@@ -195,6 +204,23 @@ function install_kourier() {
+  kubectl patch clusterrolebinding ambassador -p '{"subjects":[{"kind": "ServiceAccount", "name": "ambassador", "namespace": "ambassador"}]}' || return 1
+
+  echo ">> Enabling Knative support in Ambassador"
+  kubectl set env --namespace ambassador deployments/ambassador AMBASSADOR_KNATIVE_SUPPORT=true || return 1
```
```suggestion
  kubectl set env --namespace ambassador deployments/ambassador AMBASSADOR_KNATIVE_SUPPORT=true
```

mattmoor replied:
Can we fold these alterations into the checked in manifest?

mattmoor commented:
> Run ./test/e2e-tests.sh --ambassador-version latest.

How do we validate the Prow integration?  Do we YOLO this in, create the Prow config, and then fix things up post-submit?  Do we create the leg now without triggers, then `/test` it?  What are the next steps?

adrcunha commented:
> > Run ./test/e2e-tests.sh --ambassador-version latest.
> 
> How do we validate the Prow integration? Do we YOLO this in, create the Prow config, and then fix things up post-submit?

Yes. Merge this, then merge #test-infra/1598.

tcnghia commented:
/approve
